extensions/src/main/java/org/pentaho/platform/plugin/action/mondrian/catalog/MondrianCatalogHelper.java

```diff
@@ -600,11 +600,8 @@ public synchronized void addCatalog( InputStream schemaInputStream, final Mondri
       MondrianCatalogHelper.logger.debug( "addCatalog" ); //$NON-NLS-1$
     }
 
-    if ( isUsingRepository() ) {
-      loadCatalogIntoCache( catalog.getName(), pentahoSession );
-    } else {
-      init( pentahoSession );
-    }
+    // if cache was not initialized, do so now
+    init( pentahoSession );
```

Comment:
If this only initialises when not initialised, why not reflect that in the method name?

Reply:
Method renamed and visibility changed protected -> private